editor/src/clj/editor/camera.clj

```diff
@@ -384,6 +384,17 @@
     (assoc (camera-move camera (.x delta) (.y delta) (.z delta))
       :focus-point (doto focus (.add delta)))))
 
+(defn pan-at-pointer-position
+  [^Camera camera ^Camera prev-camera ^Region viewport [^double x ^double y]]
+  (let [focus ^Vector4d (:focus-point camera)
+        point (camera-project camera viewport (Point3d. (.x focus) (.y focus) (.z focus)))
+        screen-z (.z point)
+        world (camera-unproject camera viewport x y screen-z)
+        delta (camera-unproject prev-camera viewport x y screen-z)]
+    (.sub delta world)
+    (assoc (camera-move camera (.x delta) (.y delta) (.z delta))
+           :focus-point (doto focus (.add delta)))))
+
```

Comment:
Panning depends on the diff of the current camera, and the camera before calling `dolly`.